### PR TITLE
Fix adding/updating access codes.

### PR DESCRIPTION
**Pull request by rockymountainnative**
Add accessCodeLength and notificationEnabled. accesscodeId is the only value returned when updating/adding code.

### CODE REVIEW

**Comment by dknowles2** [comment]
Can you update the tests?

**Comment by rockymountainnative** [comment]
Unit tests have been updated. I removed one assertion. I was unable to test the api being able to pick up any out-of-band updates. The response from the api was returning either `{ "accesscodeId": "XXX" }` or `{}`. No other fields have been returned which leads me to believe the API isn't intended to return the full object fields or updated fields.

A couple other call outs:

I am unclear as to what "notification" even represents. In all of my codes, both configured to notify and not notify, all responses from the getaccesscodes has been 0. It appears when adding/updating an access code, it is under `notificationEnabled` reflected in this PR.

I am not fully up to speed on notifications within the API and left it untouched. In some cases in the app, the notificationId was included in the update/add POST command, but other times it was not. I am leaving this detail out of scope for this PR.

File: tests/test_code.py [hunk 1]
```diff
@@ -89,8 +89,6 @@ def test_save(
 
         new_json = deepcopy(access_code_json)
```
[the remark this answers is not in the document]
Comment (rockymountainnative) [reply tests/test_code.py]: This is updated.

File: pyschlage/code.py [hunk 2]
```diff
@@ -245,7 +247,11 @@ def save(self):
 
         command = "updateaccesscode" if self.access_code_id else "addaccesscode"
         resp = self._device.send_command(command, self.to_json())
-        self._update_with(self._device, resp.json())
+
+        resp_json = resp.json()
```
[the remark this answers is not in the document]
Comment (rockymountainnative) [reply pyschlage/code.py:255]: Comment added to reflect why self._updated_with() was not called. Calling out that the alternate path would have been to modify from_json() to no longer require fields that do not come back from the response. I was torn here, but we are dealing with effectively two contracts (1, from the get call and the 2 from the commands POST response.)

File: pyschlage/code.py [hunk 3]
```diff
@@ -216,7 +216,9 @@ def to_json(self) -> dict:
         json = {
             "friendlyName": self.name,
             "accessCode": int(self.code),
+            "accessCodeLength": len(str(self.code)),
```
[the remark this answers is not in the document]
Comment (rockymountainnative) [reply pyschlage/code.py]: Thanks for catching this

File: pyschlage/code.py [hunk 4]
```diff
@@ -194,14 +194,14 @@ def from_json(cls, auth: Auth, device: Device, json: dict[str, Any]) -> AccessCo
         else:
             schedule = TemporarySchedule.from_json(json)
 
+        access_code_length = json.get("accessCodeLength") or 4
```
[the remark this answers is not in the document]
Comment (rockymountainnative) [reply pyschlage/code.py]: Thanks!

**Review by dknowles2 — APPROVED** [review]

**Comment by dknowles2** [comment]
Thanks!

**Pull request merged**